src/validate.ts

```diff
@@ -32,3 +36,27 @@ export function validateDependencies (ctx: BuildContext) {
     consola.warn('Potential implicit dependencies found:', Array.from(implicitDependnecies).map(id => chalk.cyan(id)).join(', '))
   }
 }
+
+export function validateBuildOutputs (pkg: PackageJson, rootDir: string) {
+  if (!pkg) { return }
+
+  const filenames = new Set([
+    ...typeof pkg.bin === 'string' ? [pkg.bin] : Object.values(pkg.bin || {}),
+    pkg.main,
+    pkg.module,
+    pkg.types,
+    pkg.typings,
+    ...extractExportFilenames(pkg.exports).map(i => i.file)
+  ].map(i => i && resolve(rootDir, i.replace(/\/[^/]*\*.*$/, ''))))
+
+  const missingOutputs = []
+
+  for (const filename of filenames) {
+    if (filename && !filename.includes('*') && !existsSync(filename)) {
+      missingOutputs.push(filename.replace(rootDir + '/', ''))
+    }
+  }
+  if (missingOutputs.length) {
+    consola.warn(`Potential missing build outputs: ${missingOutputs.map(o => chalk.cyan(o)).join(', ')}`)
```

Comment:
Package files are not necessarility build artifacts:

```suggestion
    consola.warn(`Potential missing package.json files: ${missingOutputs.map(o => chalk.cyan(o)).join(', ')}`)
```